### PR TITLE
[stable/oauth2-proxy] add htpasswd-file components

### PR DESCRIPTION
**Pull request by dashford**
#### What this PR does / why we need it:

This PR allows us to use the `htpasswd-file` argument for the proxy. This is useful in cases where two services need to talk to each other that don't have an account with one of the providers e.g. Grafana querying a Prometheus instance which is protected by the oauth proxy.

#### Which issue this PR fixes

n/a

#### Special notes for your reviewer:

n/a

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] title of the PR contains starts with chart name e.g. `[stable/chart]`

### CODE REVIEW

**Comment by dashford** [comment]
@viglesiasce @Miouge1 does this approach look ok to you?

File: stable/oauth2-proxy/templates/deployment.yaml [hunk 1]
```diff
@@ -149,6 +157,14 @@ spec:
 {{- end }}
         name: configaccesslist
 {{- end }}
+
+{{- if .Values.htpasswdFile.enabled }}
+      - configMap:
+          defaultMode: 420
+          name: {{ template "oauth2-proxy.fullname" . }}-htpasswd-file
```
Comment (andrecastro) [inline stable/oauth2-proxy/templates/deployment.yaml]: Wouldn't be better to map this as a k8s secret? 
Maybe add something like `htpasswd-file-existing-secret` as well, would be useful

**Review by andrecastro — COMMENTED** [review]
Really good initiative! This would be very useful for my team! 
Please consider my comment.

File: stable/oauth2-proxy/templates/deployment.yaml (hunk 1 above)
Comment (dashford) [reply stable/oauth2-proxy/templates/deployment.yaml]: Hey @andrecastro, I implemented those changes. Nicer to have the file as a secret :+1:

**Comment by morganchristiansson** [comment]
I just deployed this. Works great no issues. Thank you.

**Comment by dashford** [comment]
Hi @Miouge1, would you be able to take a look at this PR? Let me know if I need to assign it to another reviewer.

**Comment by desaintmartin** [comment]
Looks good to me.
/approve

**Pull request merged**